client/src/components/JurisdictionAdmin/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,645 @@
+      }
+    }
+
+    setIsEditing(true)
+  }
+
+  const initiateSheetRename = (index: number) => {
+    setIsRenamingSheet(true)
+    setSheetNameBeforeRename({
+      index,
+      name: getValues(`resultTallySheets[${index}].name`),
+    })
+  }
+
+  const confirmSheetRename = () => {
+    setIsRenamingSheet(false)
+    setSheetNameBeforeRename(undefined)
+  }
+
+  const cancelSheetRename = () => {
+    setIsRenamingSheet(false)
+    if (sheetNameBeforeRename) {
+      const { index, name } = sheetNameBeforeRename
+      setValue(`resultTallySheets[${index}].name`, name)
+      setSheetNameBeforeRename(undefined)
+    }
+  }
+
+  const addSheet = () => {
+    let newSheetName = defaultResultTallySheetName(sheets.length)
+    function doesNewSheetNameConflict() {
+      return sheets.some(
+        (_, i) => getValues(`resultTallySheets[${i}].name`) === newSheetName
+      )
+    }
+
+    let incrementToAvoidConflict = 0
+    while (doesNewSheetNameConflict()) {
+      incrementToAvoidConflict += 1
+      newSheetName = defaultResultTallySheetName(
+        sheets.length + incrementToAvoidConflict
+      )
+    }
+
+    appendSheet({
+      name: newSheetName,
+      results: {},
+    })
+    setSelectedTabId(newSheetName)
+  }
+
+  const deleteSheet = (index: number) => {
+    const numSheetsAfterDeletion = sheets.length - 1
+
+    if (numSheetsAfterDeletion === 1 || index === 0) {
+      setSelectedTabId(VOTE_TOTALS_TAB_ID)
+    } else {
+      // Open the sheet before the to-be-deleted sheet
+      setSelectedTabId(
+        getValues(`resultTallySheets[${Math.max(index - 1, 0)}].name`)
+      )
+    }
+
+    // If we drop back down to 1 sheet, rename that sheet "Sheet 1" under the hood
+    if (numSheetsAfterDeletion === 1) {
+      const remainingIndex = index === 1 ? 0 : 1
+      setValue(`resultTallySheets[${remainingIndex}].name`, 'Sheet 1')
+    }
+
+    removeSheet(index)
+  }
+
+  const saveResults = (e: BaseSyntheticEvent) => {
+    cancelSheetRename()
+    setIsEditing(false)
+    handleSubmit(onValidSubmit, onInvalidSubmit)(e)
+  }
+
+  // ---------- Handlers (END) ----------
+
+  return (
+    <ListAndDetail>
+      <List
+        search={{
+          placeholder: 'Search batches...',
+          setQuery: setSearchQuery,
+        }}
+      >
+        {filteredBatches.map(batch => (
+          <ListItem
+            key={batch.id}
+            onClick={() => selectBatch(batch.id)}
+            selected={batch.id === selectedBatchId}
+          >
+            {batch.name}
+          </ListItem>
+        ))}
+      </List>
+
+      <Detail>
+        {!selectedBatch ? (
+          <p>Select a batch to enter audit results.</p>
+        ) : (
+          <>
+            <H4>{selectedBatch.name}</H4>
+            <Tabs
+              id={selectedBatch.name}
+              onChange={(newTabId: string) => {
+                cancelSheetRename()
+                setSelectedTabId(newTabId)
```

Comment:
Is it possible to remove the blue focus outline that happens when you click on a tab or on a batch list item?

Reply:
This was a really interesting one! We need the blue focus outline (or some indication in general) for accessible tab navigation, but we don't need/want it for mouse navigation. I initially looked at using the `:focus-visible` selector, which allows you to distinguish focus by keyboard vs. focus by mouse.

But then discovered that this is a common enough wish that Blueprint provides [a utility](https://blueprintjs.com/docs/versions/3/#core/accessibility.focus-management) for it! We'll now get the desired behavior globally.

https://github.com/votingworks/arlo/pull/1644/commits/4aaf474beb1ba2a02764e5983b20b9ad2cd56449